### ISSUE TITLE
[Question/Bug] Test using POM opens blank page instead of use fixture: one account per parallel worker

### ISSUE THREAD

**Issue by Malgosiaaa** [microsoft/playwright#27213]
Playwright Version: 1.38.0
OS version: Mac Ventura 13.5

Hello!
In my tests I need to log in as different user per worker, because tests will modify something on server side. Already tried to use one user account but as I thought there were interferences between workers what was causing flakiness/failures. Login scenario follow steps:

1.  Log in to admin panel
2. Search for different user account per worker
3. Enter to searched user view in admin - different per worker
4. Click _Log in as user_ what opens new tab with different user account logged in per every worker. To make things simplified I removed 'target' from _Log in as user_ button in order to open authenticated application in the same tab as admin panel.

I followed this documentation: [Moderate: one account per parallel worker.](https://playwright.dev/docs/auth#moderate-one-account-per-parallel-worker) and this is my fixture:

```ts
import { test as baseTest, expect } from '@playwright/test';
import fs from 'fs';
import path from 'path';

export * from '@playwright/test';
export const test = baseTest.extend<{}, { workerStorageState: string }>({
  // Use the same storage state for all tests in this worker.
  storageState: ({ workerStorageState }, use) => use(workerStorageState),

  // Authenticate once per worker with a worker-scoped fixture.
  workerStorageState: [async ({ browser }, use) => {
    // Use parallelIndex as a unique identifier for each worker.
    const id = test.info().parallelIndex;
    const fileName = path.resolve(test.info().project.outputDir, `.auth/${id}.json`);

    if (fs.existsSync(fileName)) {
      // Reuse existing authentication state if any.
      await use(fileName);
      return;
    }

    // Important: make sure we authenticate in a clean environment by unsetting storage state.
    const page = await browser.newPage({ storageState: undefined});

    // Define Interface with UserCredentials
    interface UserCredentials {
        login: string;
        pass: string;
        account_id: string;
        email: string;
      }
      
     // Simulated user data
     const userCredentialsMap: Record<number, UserCredentials> = {
        0: { login: 'test1@test1.com', pass: 'Test1', account_id: '1', email: 'test1@test1.com'},
        1: { login: 'test2@test2.com', pass: 'Test2', account_id: '2', email: 'test2@test2.com'},
        2: { login: 'test3@test3.com', pass: 'Test3', account_id: '3', email: 'test3@test3.com'},
     };

    // Log in to admin
    await page.goto(process.env.BASE_URL as string);
    await page.waitForURL(process.env.BASE_URL as string);
    await page.getByPlaceholder('Email').fill(userCredentialsMap[id].login);
    await page.getByPlaceholder('Password').fill(userCredentialsMap[id].pass);
    await page.getByText('Sign in').click();
    
    // Wait until the page receives the cookies.
    await page.waitForURL('https://tested-url-admin.com');
    
    // Search and enter to different user per every worker
    await page.getByPlaceholder('Search').fill(userCredentialsMap[id].account_id);
    await page.getByPlaceholder('Search').press('Enter');
    await page.getByText(userCredentialsMap[id].email).click();
    
    // Open separate user account in every worker, remove 'target' in order to open it in the same tab
    await page.$eval('div#root button > a', el => el.removeAttribute('target'));
    await page.getByText('Log in as user').click()
    await page.waitForLoadState();
    await expect(page).toHaveURL('https://tested-url-dashboard.com');

    // End of authentication steps.

    await page.context().storageState({ path: fileName });
    await page.close();
    await use(fileName);
  }, { scope: 'worker' }],
});

```

Above fixture works and every worker enters to application using different user account. 

**The issue is:**

1. Before test execution page which is correctly authenticated thanks to the above fixture closes. 
2. New window with blank page is opened and Playwright tries to perform test steps in that blank page and fails. 

This is the test, it uses Page Object Model:

```ts

import { test } from '../fixtures/authenticate-per-worker-fixture';
import { DashboardPage } from '../page-object-models/dashboard-page';

test('Create new item', async ({ page }) => {
        const dashboardPage = new DashboardPage(page);
        await dashboardPage.createNewItem('test.com', 'test item name');
    });


```

And this is the config file:

```ts

import { defineConfig, devices } from '@playwright/test';
import * as dotenv from 'dotenv';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
dotenv.config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Maximum time one test can run for */
  timeout: 5 * 10 * 1000,
  /* Maximum time expect() should wait for condition to be met */
  expect: {
    timeout: 1000,
  },
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    baseURL: process.env.BASE_URL,

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on',
    video: 'on',
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },

    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },
    },

    {
      name: 'webkit',
      use: { ...devices['Desktop Safari'] },
    },
  ],
});

```

Maybe do you have any advice how to perform test steps in the same authenticated browser context created by fixture?

**Comment by yury-s** [comment]
> * Before test execution page which is correctly authenticated thanks to the above fixture closes.

Correct, the page is closed in the end of the fixture (`await page.close();`). This is working as intended - the logged in state (cookies and local storage) is expected to be stored in the storage state, that state will be loaded into the new context before each test.

> * New window with blank page is opened and Playwright tries to perform test steps in that blank page and fails.

This is working as expected, the test starts with an empty page in a new context, the context is populated with the storage state from the fixture and if you navigate to the admin/user page you should not need to repeat authentication steps as the cookies are already there. You can add [baseURL](https://playwright.dev/docs/api/class-testoptions#test-options-base-url) option if you want the page to navigate somewhere before each test. Can you be more specific about why the test fails and why you think this is a bug in playwright?

**Comment by yury-s** [comment]
Closing per the response above, feel free to open a new issue if it doesn't work.
